File: java/core/src/test/java/dev/varar/core/ParseTest.java
package dev.varar.core;

import static org.junit.jupiter.api.Assertions.assertEquals;

import dev.varar.core.Ast.Doc;
import java.util.List;
import org.junit.jupiter.api.Test;

/** Port of {@code typescript/packages/core/tests/parse.test.ts}. */
class ParseTest {

    @Test
    void parseReturnsADocWhoseExamplesComeFromParagraphsAndCarryTheHeadingStack() {
        String source = "# Hello\n\nbody";
        Doc doc = Parse.parse("hello.md", source);
        assertEquals("hello.md", doc.path());
        assertEquals(source, doc.source());
        // One paragraph, one Example. Example name is computed by the planner, not captured
        // here; the structurer's job is just to track scope + body.
        assertEquals(1, doc.examples().size());
        assertEquals(List.of("Hello"), doc.examples().get(0).scopeStack());
    }
}

File: rust/core/tests/parse_test.rs
//! Port of `ParseTest.java` / `parse.test.ts`.

use varar_core::parse::parse;

#[test]
fn parse_returns_a_doc_whose_examples_come_from_paragraphs_and_carry_the_heading_stack() {
    let source = "# Hello\n\nbody";
    let doc = parse("hello.md", source);
    assert_eq!("hello.md", doc.path);
    assert_eq!(source, doc.source);
    assert_eq!(1, doc.examples.len());
    assert_eq!(vec!["Hello".to_string()], doc.examples[0].scope_stack);
}
